app/multitenant/dynamo_collector.go

```diff
@@ -330,13 +386,35 @@ func (c *dynamoDBCollector) getReports(userid string, row int64, start, end time
 		return nil, err
 	}
 
+	// TODO(jml): Refactor this so that we have a standard interface
+	// IReportStore for "fetch :: [ReportKeys] -> ([Report], missing, error)"
+	// and a list of providers: in-memory cache, memcache, dynamo. Possibly
+	// also an "always everything missing" implementation to simplify logic.
 	cachedReports, missing := c.getCached(reportKeys)
 	dynamoCacheHits.Add(float64(len(cachedReports)))
 	dynamoCacheMiss.Add(float64(len(missing)))
 	if len(missing) == 0 {
 		return cachedReports, nil
 	}
 
+	if c.memcache != nil {
+		var memcachedReports []report.Report
+		memcachedReports, missing, err = c.fetchFromMemcache(missing)
+		memcacheHits.Add(float64(len(memcachedReports)))
+		memcacheMiss.Add(float64(len(missing)))
+		if err != nil {
+			// REVIEWER: jml is unclear whether we should abort in this case or
+			// just carry on. Aborting is easier to reason about for us, but
+			// suppressing is probably OK since failure just means we fetch
+			// from S3.
```

Comment:
* fail to successfully talk to memcache: dodgy network, dodgy server. Monitoring should catch this
* corrupt report in the memcache: we have a warning for this but no monitoring

I'll delete the comment and leave it erroring for now.